segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java

```diff
@@ -69,13 +69,14 @@
      * @param streamSegmentStore  The SegmentStore to delegate all requests to.
      * @param tableStore          The SegmentStore to delegate all requests to.
      * @param tokenExpiryExecutor The executor to be used for running token expiration handling tasks.
+     * @param tlsProtocolVersion the version of the TLS protocol
      */
     @VisibleForTesting
     public PravegaConnectionListener(boolean enableTls, int port, StreamSegmentStore streamSegmentStore,
-                                     TableStore tableStore, ScheduledExecutorService tokenExpiryExecutor) {
+                                     TableStore tableStore, ScheduledExecutorService tokenExpiryExecutor, String tlsProtocolVersion) {
```

Comment:
Is there anything we can do to prevent this constructor from growing even more?

Reply:
@andreipaduroiu ; agreed, I had mentioned that in point 1 of https://github.com/pravega/pravega/pull/6114#pullrequestreview-697598121

One thing I had suggested to Sai is to make call to either address this additional refactoring in this PR or create a separate issue and handle it there.   It IS kind of a separate issue, BUT we happen to be stepping right on it here by making these changes.

Reply:
My personal preference would be to create a separate issue for the refactoring of all the settings that are getting passed to the PravegaConnectionListener and leave this PR focused on what it's currently doing.

This will allow us to proceed with what this PR has to give and take the time to discuss the perfect options of the constructor (multiple constructore, factory, builders etc.) as a separate effort.  Thoughts?

Reply:
IntelliJ refactor can help in extracting Builder which can be used to construct PravegaConnectionListener that has some mandatory and some optional parameters.

Reply:
I've created a constructor for `pravegaconnectionlistener` (constructor chaining) which reduces the no of test files.